diff --git a/README.md b/README.md
index 044c0d4..f373282 100644
--- a/README.md
+++ b/README.md
@@ -9,7 +9,13 @@ Open this directory in a command prompt, then:
 2. Run the development server: `python manage.py runserver`
 3. Go to http://localhost:8000/ in your browser, or http://localhost:8000/admin/ to log in and get to work!
 
-## Documentation links
+## Installing in Kubernetes
+
+    cat values.yaml | sed "s#%%SENTRY_DSN%%#<VAR_SENTRY_DSN>#g" | \
+    helm upgrade web --namespace verypythonproblems --install deploy -f -
+
+
+## CodeRed CMS Documentation links
 
 * To customize the content, design, and features of the site see [CodeRed CMS](https://docs.coderedcorp.com/cms/).
 * For deeper customization of backend code see [Wagtail](http://docs.wagtail.io/) and [Django](https://docs.djangoproject.com/).
diff --git a/deploy/templates/deployment.yaml b/deploy/templates/deployment.yaml
index 663de4d..5beb5ee 100644
--- a/deploy/templates/deployment.yaml
+++ b/deploy/templates/deployment.yaml
@@ -25,6 +25,8 @@ spec:
         - name: {{ .Chart.Name }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            {{- toYaml .Values.environment.variables | nindent 12 }}
           ports:
             - name: container-port
               containerPort: {{ .Values.service.port }}
@@ -40,6 +42,7 @@ spec:
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
       {{- with .Values.nodeSelector }}
+      # ENVIRONMENT VARIABLE FOR SENTRY.
       nodeSelector:
         {{- toYaml . | nindent 8 }}
       {{- end }}
diff --git a/deploy/values.yaml b/deploy/values.yaml
index d1e4a73..78d6e62 100644
--- a/deploy/values.yaml
+++ b/deploy/values.yaml
@@ -8,7 +8,8 @@ image:
   repository: nginx
   tag: stable
   pullPolicy: IfNotPresent
-
+environment:
+  variables: []
 imagePullSecrets: []
 nameOverride: ""
 fullnameOverride: ""
diff --git a/requirements.txt b/requirements.txt
index fbec82e..f05afa8 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -7,3 +7,4 @@ django_sendmail_backend
 # To use with a MySQL database (offered by most web hosts), uncomment the following.
 #mysqlclient
 django-prometheus
+sentry-sdk==0.9.0
diff --git a/values.yaml b/values.yaml
index 3045b96..6110d3c 100644
--- a/values.yaml
+++ b/values.yaml
@@ -2,8 +2,12 @@
 fullnameOverride: "web"
 image:
   repository: p6rguvyrst/demo-instrumentation
-  tag: v0.1.3
+  tag: v0.1.4
   pullPolicy: Always
+environment:
+  variables:
+    - name: SENTRY_DSN
+      value: "%%SENTRY_DSN%%"
 service:
   type: ClusterIP
   port: 8000
diff --git a/verypythonproblems/exceptions/__init__.py b/verypythonproblems/exceptions/__init__.py
new file mode 100644
index 0000000..9f86505
--- /dev/null
+++ b/verypythonproblems/exceptions/__init__.py
@@ -0,0 +1 @@
+from .base import VeryPythonException
diff --git a/verypythonproblems/exceptions/base.py b/verypythonproblems/exceptions/base.py
new file mode 100644
index 0000000..9e4b0ea
--- /dev/null
+++ b/verypythonproblems/exceptions/base.py
@@ -0,0 +1,2 @@
+class VeryPythonException(Exception):
+    pass
diff --git a/verypythonproblems/features.py b/verypythonproblems/features.py
new file mode 100644
index 0000000..5b951d4
--- /dev/null
+++ b/verypythonproblems/features.py
@@ -0,0 +1,5 @@
+from .exceptions import VeryPythonException
+
+
+def trigger_error(request):
+    raise VeryPythonException("When things go wrong...")
diff --git a/verypythonproblems/settings/base.py b/verypythonproblems/settings/base.py
index 84a4ed2..fc665cc 100644
--- a/verypythonproblems/settings/base.py
+++ b/verypythonproblems/settings/base.py
@@ -12,10 +12,18 @@ https://docs.djangoproject.com/en/2.1/ref/settings/
 
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 import os
+import sentry_sdk
+from sentry_sdk.integrations.django import DjangoIntegration
 
 PROJECT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 BASE_DIR = os.path.dirname(PROJECT_DIR)
+SENTRY_DSN = os.environ.get('SENTRY_DSN', None)
 
+if SENTRY_DSN:
+    sentry_sdk.init(
+        dsn=SENTRY_DSN,
+        integrations=[DjangoIntegration()]
+    )
 
 # Quick-start development settings - unsuitable for production
 # See https://docs.djangoproject.com/en/2.1/howto/deployment/checklist/
diff --git a/verypythonproblems/urls.py b/verypythonproblems/urls.py
index bdc026f..94b57df 100644
--- a/verypythonproblems/urls.py
+++ b/verypythonproblems/urls.py
@@ -5,6 +5,8 @@ from wagtail.documents import urls as wagtaildocs_urls
 from coderedcms import admin_urls as coderedadmin_urls
 from coderedcms import search_urls as coderedsearch_urls
 from coderedcms import urls as codered_urls
+from verypythonproblems.features import trigger_error
+
 
 urlpatterns = [
     # Admin
@@ -19,6 +21,7 @@ urlpatterns = [
 
     # Instrumentation
     path('', include('django_prometheus.urls')),
+    path('sentry-debug/', trigger_error),
 
     # For anything not caught by a more specific rule above, hand over to
     # the page serving mechanism. This should be the last pattern in
