diff --git a/values.yaml b/values.yaml
index 74fa896..2c275a0 100644
--- a/values.yaml
+++ b/values.yaml
@@ -2,7 +2,7 @@
 fullnameOverride: "web"
 image:
   repository: p6rguvyrst/demo-instrumentation
-  tag: v0.1.5
+  tag: v0.1.6
   pullPolicy: Always
 environment:
   variables:
diff --git a/verypythonproblems/urls.py b/verypythonproblems/urls.py
index 94b57df..f0ccc45 100644
--- a/verypythonproblems/urls.py
+++ b/verypythonproblems/urls.py
@@ -5,7 +5,9 @@ from wagtail.documents import urls as wagtaildocs_urls
 from coderedcms import admin_urls as coderedadmin_urls
 from coderedcms import search_urls as coderedsearch_urls
 from coderedcms import urls as codered_urls
-from verypythonproblems.features import trigger_error
+from verypythonproblems.features import (
+    trigger_error,
+)
 
 
 urlpatterns = [
@@ -21,7 +23,7 @@ urlpatterns = [
 
     # Instrumentation
     path('', include('django_prometheus.urls')),
-    path('sentry-debug/', trigger_error),
+    path('sentry-debug', trigger_error),
 
     # For anything not caught by a more specific rule above, hand over to
     # the page serving mechanism. This should be the last pattern in
diff --git a/website/models.py b/website/models.py
index 7734c14..2e09824 100644
--- a/website/models.py
+++ b/website/models.py
@@ -10,9 +10,10 @@ from coderedcms.models import (
     CoderedFormPage,
     CoderedWebPage
 )
+from django_prometheus.models import ExportModelOperationsMixin
 
 
-class ArticlePage(CoderedArticlePage):
+class ArticlePage(ExportModelOperationsMixin('article'), CoderedArticlePage):
     """
     Article, suitable for news or blog content.
     """
@@ -28,7 +29,7 @@ class ArticlePage(CoderedArticlePage):
     search_template = 'coderedcms/pages/article_page.search.html'
 
 
-class ArticleIndexPage(CoderedArticleIndexPage):
+class ArticleIndexPage(ExportModelOperationsMixin('article-index'), CoderedArticleIndexPage):
     """
     Shows a list of article sub-pages.
     """
@@ -44,7 +45,7 @@ class ArticleIndexPage(CoderedArticleIndexPage):
     template = 'coderedcms/pages/article_index_page.html'
 
 
-class FormPage(CoderedFormPage):
+class FormPage(ExportModelOperationsMixin('form'), CoderedFormPage):
     """
     A page with an html <form>.
     """
@@ -54,7 +55,7 @@ class FormPage(CoderedFormPage):
     template = 'coderedcms/pages/form_page.html'
 
 
-class FormPageField(CoderedFormField):
+class FormPageField(ExportModelOperationsMixin('form-field'), CoderedFormField):
     """
     A field that links to a FormPage.
     """
@@ -63,14 +64,14 @@ class FormPageField(CoderedFormField):
 
     page = ParentalKey('FormPage', related_name='form_fields')
 
-class FormConfirmEmail(CoderedEmail):
+class FormConfirmEmail(ExportModelOperationsMixin('email'), CoderedEmail):
     """
     Sends a confirmation email after submitting a FormPage.
     """
     page = ParentalKey('FormPage', related_name='confirmation_emails')
 
 
-class WebPage(CoderedWebPage):
+class WebPage(ExportModelOperationsMixin('webpage'), CoderedWebPage):
     """
     General use page with featureful streamfield and SEO attributes.
     Template renders all Navbar and Footer snippets in existance.
