diff --git a/charts/incubator/wagtail/templates/deployment.yaml b/charts/incubator/wagtail/templates/deployment.yaml
index 53de42d..0ebb993 100644
--- a/charts/incubator/wagtail/templates/deployment.yaml
+++ b/charts/incubator/wagtail/templates/deployment.yaml
@@ -24,6 +24,8 @@ spec:
         - name: {{ .Chart.Name }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            {{- toYaml .Values.environment.variables | nindent 12 }}
           ports:
             - name: container-port
               containerPort: {{ .Values.service.port }}
diff --git a/charts/incubator/wagtail/values.yaml b/charts/incubator/wagtail/values.yaml
index ba4cba9..0b5c888 100644
--- a/charts/incubator/wagtail/values.yaml
+++ b/charts/incubator/wagtail/values.yaml
@@ -17,6 +17,9 @@ service:
   type: ClusterIP
   port: 80
 
+environment:
+  variables: []
+
 ingress:
   enabled: false
   annotations: {}
diff --git a/verypythonproblems/requirements.txt b/verypythonproblems/requirements.txt
index 34ae1b9..d6f0e8a 100644
--- a/verypythonproblems/requirements.txt
+++ b/verypythonproblems/requirements.txt
@@ -1,6 +1,7 @@
 coderedcms
 django-prometheus
 prometheus_client
+sentry-sdk==0.9.0
 
 # django_sendmail_backend enables sending email from your web host server.
 # Remove this if using a different email backend.
diff --git a/verypythonproblems/values.yaml b/verypythonproblems/values.yaml
index 78b42cc..39680a8 100644
--- a/verypythonproblems/values.yaml
+++ b/verypythonproblems/values.yaml
@@ -2,11 +2,15 @@
 fullnameOverride: "web"
 image:
   repository: p6rguvyrst/instrumentation-demo
-  tag: v0.0.8
+  tag: v0.0.9
   pullPolicy: Never
 service:
   type: NodePort
   port: 8000
+environment:
+  variables:
+    - name: SENTRY_DSN
+      value: "%%SENTRY_DSN%%"
 #resources:
 #  limits:
 #    cpu: 200m
diff --git a/verypythonproblems/verypythonproblems/exceptions/__init__.py b/verypythonproblems/verypythonproblems/exceptions/__init__.py
new file mode 100644
index 0000000..9f86505
--- /dev/null
+++ b/verypythonproblems/verypythonproblems/exceptions/__init__.py
@@ -0,0 +1 @@
+from .base import VeryPythonException
diff --git a/verypythonproblems/verypythonproblems/exceptions/base.py b/verypythonproblems/verypythonproblems/exceptions/base.py
new file mode 100644
index 0000000..9e4b0ea
--- /dev/null
+++ b/verypythonproblems/verypythonproblems/exceptions/base.py
@@ -0,0 +1,2 @@
+class VeryPythonException(Exception):
+    pass
diff --git a/verypythonproblems/verypythonproblems/features.py b/verypythonproblems/verypythonproblems/features.py
new file mode 100644
index 0000000..c0eed23
--- /dev/null
+++ b/verypythonproblems/verypythonproblems/features.py
@@ -0,0 +1,5 @@
+from .exceptions import VeryPythonException
+
+
+def trigger_exception(request):
+    raise VeryPythonException("Always look on the bright side of life.")
diff --git a/verypythonproblems/verypythonproblems/settings/base.py b/verypythonproblems/verypythonproblems/settings/base.py
index f6dc8d5..1352191 100644
--- a/verypythonproblems/verypythonproblems/settings/base.py
+++ b/verypythonproblems/verypythonproblems/settings/base.py
@@ -12,9 +12,19 @@ https://docs.djangoproject.com/en/2.1/ref/settings/
 
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 import os
+import sentry_sdk
+from sentry_sdk.integrations.django import DjangoIntegration
+
 
 PROJECT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 BASE_DIR = os.path.dirname(PROJECT_DIR)
+SENTRY_DSN = os.environ.get('SENTRY_DSN', None)
+
+if SENTRY_DSN:
+    sentry_sdk.init(
+        dsn=SENTRY_DSN,
+        integrations=[DjangoIntegration()]
+    )
 
 
 # Quick-start development settings - unsuitable for production
diff --git a/verypythonproblems/verypythonproblems/urls.py b/verypythonproblems/verypythonproblems/urls.py
index 5cab977..4e6894a 100644
--- a/verypythonproblems/verypythonproblems/urls.py
+++ b/verypythonproblems/verypythonproblems/urls.py
@@ -6,6 +6,8 @@ from coderedcms import admin_urls as coderedadmin_urls
 from coderedcms import search_urls as coderedsearch_urls
 from coderedcms import urls as codered_urls
 from verypythonproblems.instrumentation.monitor import prometheus_monitor
+from verypythonproblems.features import trigger_exception
+
 
 urlpatterns = [
     # Admin
@@ -20,6 +22,7 @@ urlpatterns = [
 
     # Instrumentation
     path('metrics', prometheus_monitor, name='prometheus-django-metrics'),
+    path('sentry-debug', trigger_exception),
 
     # For anything not caught by a more specific rule above, hand over to
     # the page serving mechanism. This should be the last pattern in
