diff --git a/verypythonproblems/tests/conftest.py b/verypythonproblems/tests/conftest.py
new file mode 100644
index 0000000..e69de29
diff --git a/verypythonproblems/tests/test_instrumentation_prometheus_monitor.py b/verypythonproblems/tests/test_instrumentation_prometheus_monitor.py
new file mode 100644
index 0000000..140ef74
--- /dev/null
+++ b/verypythonproblems/tests/test_instrumentation_prometheus_monitor.py
@@ -0,0 +1,3 @@
+
+def test_models_gauge_monitor():
+    pass
diff --git a/verypythonproblems/values.yaml b/verypythonproblems/values.yaml
index 318020c..3a579a9 100644
--- a/verypythonproblems/values.yaml
+++ b/verypythonproblems/values.yaml
@@ -2,7 +2,7 @@
 fullnameOverride: "web"
 image:
   repository: p6rguvyrst/instrumentation-demo
-  tag: v0.0.9
+  tag: v0.1.0
   pullPolicy: Never
 service:
   type: NodePort
diff --git a/verypythonproblems/verypythonproblems/instrumentation/prometheus/monitor.py b/verypythonproblems/verypythonproblems/instrumentation/prometheus/monitor.py
index b4a1e8f..d0ba601 100644
--- a/verypythonproblems/verypythonproblems/instrumentation/prometheus/monitor.py
+++ b/verypythonproblems/verypythonproblems/instrumentation/prometheus/monitor.py
@@ -1,11 +1,11 @@
 from .metrics import ModelsGauges
 from ..query_sets import (
-    ARTICLE_PAGE_COUNT,
-    ARTICLE_INDEX_PAGE_COUNT,
-    FORM_PAGE_COUNT,
-    FORM_PAGE_FIELD_COUNT,
-    FORM_CONFIRM_EMAIL_COUNT,
-    WEB_PAGE_COUNT,
+    ARTICLE_PAGE,
+    ARTICLE_INDEX_PAGE,
+    FORM_PAGE,
+    FORM_PAGE_FIELD,
+    FORM_CONFIRM_EMAIL,
+    WEB_PAGE,
 )
 from ..labels import (
     ARTICLE,
@@ -20,9 +20,9 @@ GAUGE = ModelsGauges()
 
 def models_gauge_monitor():
 
-    GAUGE.django_model_objects.labels(ARTICLE).set(ARTICLE_PAGE_COUNT)
-    GAUGE.django_model_objects.labels(ARTICLE_INDEX).set(ARTICLE_INDEX_PAGE_COUNT)
-    GAUGE.django_model_objects.labels(FORM).set(FORM_PAGE_COUNT)
-    GAUGE.django_model_objects.labels(FORM_FIELD).set(FORM_PAGE_FIELD_COUNT)
-    GAUGE.django_model_objects.labels(EMAIL).set(FORM_CONFIRM_EMAIL_COUNT)
-    GAUGE.django_model_objects.labels(WEBPAGE).set(WEB_PAGE_COUNT)
+    GAUGE.django_model_objects.labels(ARTICLE).set(ARTICLE_PAGE.count())
+    GAUGE.django_model_objects.labels(ARTICLE_INDEX).set(ARTICLE_INDEX_PAGE.count())
+    GAUGE.django_model_objects.labels(FORM).set(FORM_PAGE.count())
+    GAUGE.django_model_objects.labels(FORM_FIELD).set(FORM_PAGE_FIELD.count())
+    GAUGE.django_model_objects.labels(EMAIL).set(FORM_CONFIRM_EMAIL.count())
+    GAUGE.django_model_objects.labels(WEBPAGE).set(WEB_PAGE.count())
diff --git a/verypythonproblems/verypythonproblems/instrumentation/query_sets.py b/verypythonproblems/verypythonproblems/instrumentation/query_sets.py
index 6858ba1..a2f841f 100644
--- a/verypythonproblems/verypythonproblems/instrumentation/query_sets.py
+++ b/verypythonproblems/verypythonproblems/instrumentation/query_sets.py
@@ -25,9 +25,9 @@ object.count(): https://docs.djangoproject.com/en/2.1/ref/models/querysets/#djan
 
 """
 
-ARTICLE_PAGE_COUNT = ArticlePage.objects.count()
-ARTICLE_INDEX_PAGE_COUNT = ArticleIndexPage.objects.count()
-FORM_PAGE_COUNT = FormPage.objects.count()
-FORM_PAGE_FIELD_COUNT = FormPageField.objects.count()
-FORM_CONFIRM_EMAIL_COUNT = FormConfirmEmail.objects.count()
-WEB_PAGE_COUNT = WebPage.objects.count()
+ARTICLE_PAGE = ArticlePage.objects
+ARTICLE_INDEX_PAGE = ArticleIndexPage.objects
+FORM_PAGE = FormPage.objects
+FORM_PAGE_FIELD = FormPageField.objects
+FORM_CONFIRM_EMAIL = FormConfirmEmail.objects
+WEB_PAGE = WebPage.objects
