diff --git a/values.yaml b/values.yaml
index ad58976..60fca30 100644
--- a/values.yaml
+++ b/values.yaml
@@ -2,7 +2,7 @@
 fullnameOverride: "web"
 image:
   repository: p6rguvyrst/demo-instrumentation
-  tag: v0.1.7
+  tag: v0.1.8
   pullPolicy: Always
 environment:
   variables:
diff --git a/verypythonproblems/features.py b/verypythonproblems/features.py
index ac2ecf2..1e28dae 100644
--- a/verypythonproblems/features.py
+++ b/verypythonproblems/features.py
@@ -2,8 +2,8 @@ from .exceptions import VeryPythonException
 from .instrumentation.prometheus.monitor import (
    models_gauge_monitor,
 )
-from django.http import HttpResponse
-import prometheus_client
+import django_prometheus
+
 
 def trigger_error(request):
     raise VeryPythonException("When things go wrong...")
@@ -11,6 +11,4 @@ def trigger_error(request):
 
 def custom_instrumentation_monitor(request):
     models_gauge_monitor()
-    return HttpResponse(
-        prometheus_client.generate_latest(), content_type="text/plain"
-    )
+    return django_prometheus.exports.ExportToDjangoView(request)
diff --git a/verypythonproblems/urls.py b/verypythonproblems/urls.py
index 7aaa343..1fd9cc4 100644
--- a/verypythonproblems/urls.py
+++ b/verypythonproblems/urls.py
@@ -23,9 +23,8 @@ urlpatterns = [
     path('search/', include(coderedsearch_urls)),
 
     # Instrumentation
-    path('', include('django_prometheus.urls')),
     path('sentry-debug', trigger_error),
-    path('instrumentation', custom_instrumentation_monitor),
+    path('metrics', custom_instrumentation_monitor, name='prometheus-django-metrics'),
 
     # For anything not caught by a more specific rule above, hand over to
     # the page serving mechanism. This should be the last pattern in
